### PR TITLE
fix: Upgrade argo-ui to fix intermediate parameters window. Fixes #11448

### LINKED ISSUE #11448: UI: Intermediate parameters window is not scrollable
# Summary

For intermediate parameter suspending nodes with many parameters, the window for inputting values does not scroll. The only way to see the input box is to zoom out in your browser, otherwise the input fields are accessible by tabbing through, but not visible. The "ok"/"cancel" buttons are also inaccessible without zooming out.

e.g. for a node with 15 inputs, I can only see 10 here:

image

# Use Cases

Any intermediate parameters with 10+ inputs would be easier to use

<details><summary>Example yaml to test</summary>

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: intermediate-params-test-
spec:
  entrypoint: suspender
  templates:
  - name: suspender
    suspend: {}
    inputs:
      parameters:
      - default: ''
        name: param-1
      - default: ''
        name: param-2
      - default: ''
        name: param-3
      - default: ''
        name: param-4
      - default: ''
        name: param-5
      - default: ''
        name: param-6
      - default: ''
        name: param-7
      - default: ''
        name: param-8
      - default: ''
        name: param-9
      - default: ''
        name: param-10
      - default: ''
        name: param-11
      - default: ''
        name: param-12
      - default: ''
        name: param-13
      - default: ''
        name: param-14
      - default: ''
        name: param-15
    outputs:
      parameters:
      - name: param-1
        valueFrom:
          supplied: {}
      - name: param-2
        valueFrom:
          supplied: {}
      - name: param-3
        valueFrom:
          supplied: {}
      - name: param-4
        valueFrom:
          supplied: {}
      - name: param-5
        valueFrom:
          supplied: {}
      - name: param-6
        valueFrom:
          supplied: {}
      - name: param-7
        valueFrom:
          supplied: {}
      - name: param-8
        valueFrom:
          supplied: {}
      - name: param-9
        valueFrom:
          supplied: {}
      - name: param-10
        valueFrom:
          supplied: {}
      - name: param-11
        valueFrom:
          supplied: {}
      - name: param-12
        valueFrom:
          supplied: {}
      - name: param-13
        valueFrom:
          supplied: {}
      - name: param-14
        valueFrom:
          supplied: {}
      - name: param-15
        valueFrom:
          supplied: {}
```

</details>

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by terrytangyuan**
Incorporates fix in https://github.com/argoproj/argo-ui/pull/418. Closes https://github.com/argoproj/argo-workflows/issues/11448

### CODE REVIEW

**Review by rbreeze — APPROVED** [review]

**Review by isubasinghe — CHANGES_REQUESTED** [review]
Build failures in the tests

**Comment by atomicman57** [comment]
Taking a look @terrytangyuan

**Comment by isubasinghe** [comment]
@terrytangyuan failing to replicate on my end, perhaps this is just a random incident. Could you push an empty commit and see what the CI does?

**Comment by agilgur5** [comment]
I was able to reproduce this locally -- did a fresh rebuild of my devcontainer with all Node 20 changes etc (directly on this commit, to be exact).

> @terrytangyuan failing to replicate on my end

@isubasinghe I wonder if on your end that might be because the [Nix Node 20 change](https://github.com/argoproj/argo-workflows/pull/11506) hasn't been merged yet? So if you're using the Nix env based off this PR, it might be inconsistent / out-of-date

**Comment by agilgur5** [comment]
For reference, pulling here [from Slack](https://cloud-native.slack.com/archives/C0510EUH90V/p1691564916962089?thread_ts=1691538680.554359&cid=C0510EUH90V) that these are all ESM related.
And also apparently `argo-ui` does not export compiled code, it exports [`src`](https://github.com/argoproj/argo-ui/blob/87d27fb1cb4f6e3ac4a49f85747e471b2efa7512/package.json#L8) TSX, which can result in discrepancies when different TS settings (such as [`esModuleInterop`](https://github.com/argoproj/argo-ui/blob/87d27fb1cb4f6e3ac4a49f85747e471b2efa7512/tsconfig.json#L9), which is causing errors as below) or versions are used (among potentially other issues).

There are a few different kinds of errors here:

1. 
    ```txt
    ERROR in ./node_modules/react-toastify/dist/react-toastify.esm.mjs 303:11-23
    Can't import the named export 'cloneElement' from non EcmaScript module (only default export is available)
     @ ./node_modules/argo-ui/src/components/notifications/notifications.tsx
     @ ./node_modules/argo-ui/src/components/index.ts
     @ ./node_modules/argo-ui/src/index.ts
     @ ./src/app/app.tsx
     @ ./src/app/index.tsx
   ```

   All errors of this format seem to be due to mixed React imports. This _might_ be fixable by force resolving `react-toastify` to its CJS `dist` file (if it has one). Otherwise this may require consistent deps (there were a few installation warnings on peerDeps) or force resolving `react` itself. 

1.
    ```txt
    ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/argo-ui/src/components/nav-bar/nav-bar.tsx
    ./node_modules/argo-ui/src/components/nav-bar/nav-bar.tsx
    [tsl] ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/argo-ui/src/components/nav-bar/nav-bar.tsx(1,9)
          TS2305: Module '"classnames"' has no exported member 'default'.
      ```
      
      ```txt
      ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/argo-ui/v2/utils/utils.tsx
    ./node_modules/argo-ui/v2/utils/utils.tsx
    [tsl] ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/argo-ui/v2/utils/utils.tsx(1,8)
          TS1259: Module '"/home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/moment/ts3.1-typings/moment"' can only be default-imported using the 'esModuleInterop' flag
      ```
      
      All errors of both of these formats are fixed by setting `"esModuleInterop": true` in `tsconfig.json`, as the error states.
      As I mentioned above, this is because `argo-ui` exports source TS instead of compiled JS and as such the `tsconfig.json` [must match](https://github.com/argoproj/argo-ui/blob/87d27fb1cb4f6e3ac4a49f85747e471b2efa7512/tsconfig.json#L9)
   
1.
    ```txt
    ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/src/app/event-sources/components/event-source-list/event-source-list.tsx
    ./src/app/event-sources/components/event-source-list/event-source-list.tsx
    [tsl] ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/src/app/event-sources/components/event-source-list/event-source-list.tsx(135,51)
          TS2349: This expression is not callable.
      Type '{ default: (...args: ArgumentArray) => string; }' has no call signatures.
    ```
    
    All errors of this format need to be changed from a namespace import (e.g. `import * as classNames from ...`) to a default import (e.g. `import classNames from ...`)
    
1. 
    ```txt
    ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/src/app/workflows/components/workflow-timeline/workflow-timeline.tsx
    ./src/app/workflows/components/workflow-timeline/workflow-timeline.tsx
    [tsl] ERROR in /home/vscode/go/src/github.com/argoproj/argo-workflows/ui/src/app/workflows/components/workflow-timeline/workflow-timeline.tsx(138,37)
          TS2349: This expression is not callable.
      Type 'typeof import("/home/vscode/go/src/github.com/argoproj/argo-workflows/ui/node_modules/moment/ts3.1-typings/moment.d.ts")' has no call signatures.
      ```
      
      Same as above, errors of this format can be fixed by using a default import: `import moment from 'moment';`

File: ui/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "node": ">=20"
     },
     "dependencies": {
-        "argo-ui": "https://github.com/argoproj/argo-ui.git#v2.5.0",
+        "argo-ui": "git+https://github.com/argoproj/argo-ui.git",
```
Comment (agilgur5) [inline ui/package.json:18]: this should ideally pin a SHA if not a tag

**Comment by agilgur5** [comment]
> 1. All errors of this format seem to be due to mixed React imports. This _might_ be fixable by force resolving `react-toastify` to its CJS `dist` file (if it has one). Otherwise this may require consistent deps (there were a few installation warnings on peerDeps) or force resolving `react` itself.

For reference, these all ended up being due to usage of old Webpack v4 here. This required force resolving `react-toastify` to `9.0.3` per https://github.com/fkhadra/react-toastify/issues/775#issuecomment-1150715526 (the issue talks about CRA, but this seems to impact (be caused by?) upstream Webpack). Notably, Argo CD [uses](https://github.com/argoproj/argo-cd/blob/bfaac2b5ac4b96c8af8158902c1b086f7ced7389/ui/package.json#L119) Webpack v5.
(In general, the UI build process could use some love and updating; I'd probably suggest moving to `esbuild` [similar to Argo CD](https://github.com/argoproj/argo-cd/blob/bfaac2b5ac4b96c8af8158902c1b086f7ced7389/ui/src/app/webpack.config.js#L37) as the build is currently _very_ slow)

Also fixed the deprecation warning from the old `foundation-sites` dep by upgrading it.

Will have a replacement PR out shortly

**Comment by agilgur5** [comment]
Fixed all of the above in #11585

**Comment by isubasinghe** [comment]
> @isubasinghe I wonder if on your end that might be because the https://github.com/argoproj/argo-workflows/pull/11506 hasn't been merged yet? So if you're using the Nix env based off this PR, it might be inconsistent / out-of-date

Oh yes that could be the reason. 

Nice work on getting the PR for this out though!

**Pull request closed without merging**